os/se/ameba/security_ameba_wrapper_tz.c

```diff
@@ -104,24 +105,40 @@ typedef struct {
 } inout_struc;
 
 typedef struct {
-	uint32_t factorykey_device_idx;
-	uint32_t factorykey_cert_idx;
 	uint32_t factory_cert_addr;
 	uint32_t factory_key_addr;
 	uint32_t sstorage_addr;
 	uint32_t samsung_key_addr;
+	hal_key_type factory_slot_key_type[USABLE_FACTORY_KEY_INDEX];
 } factory_struc;
 
 static uint8_t *ns_buf = NULL;
-
 extern void *rtl_set_ns_func(void);
 
+/**
+ * Helper function
+ */
+static int se_ameba_index_chk(uint32_t key_idx)
+{
+	uint32_t ret_key_idx = HAL_SUCCESS;
```

Comment:
can we remove this multiple checks and add only one:
if ((key_idx < USABLE_FACTORY_KEY_INDEX)
return HAL_SUCCESS;
else
return HAL_INVALID_SLOT_RANGE;

Reply:
se_ameba_index_chk() will check for RTL SE invalid slot 8 to 31.